code/modules/mob/living/living.dm

```diff
@@ -734,13 +734,29 @@ Thanks.
 	//Getting out of someone's inventory.
 	if(istype(src.loc,/obj/item/weapon/holder))
 		var/obj/item/weapon/holder/H = src.loc
-		src.loc = get_turf(src.loc)
+		forceMove(get_turf(src))
 		if(istype(H.loc, /mob/living))
 			var/mob/living/Location = H.loc
 			Location.drop_from_inventory(H)
 		qdel(H)
 		H = null
 		return
+	else if(istype(src.loc, /obj/structure/strange_present))
```

Comment:
Wow clusterfack I didn't review this because:
* I'm lazy.
* I assumed you knew better than this...